### ISSUE TITLE
mangling of CSS variables (custom properties)?

### ISSUE THREAD

**Issue by tameraydin** [sndyuk/mangle-css-class-webpack-plugin#51]
Hi, thanks for this project!

Although the focus of plugin's is clearly class-names only, it would be nice if it also mangles CSS variables (custom properties) optionally.

```css
:root {
  --l-long-variable-names: 1px;
}
... {
  padding: var(--l-long-variable-names)
}
```
->
```css
:root {
  --a: 1px;
}
... {
  padding: var(--a)
}
```

Practically below change will enable it:

```diff
-    classnameRegex = new RegExp(`\\\.(${opts.classNameRegExp})`, 'g');
+    classnameRegex = new RegExp(`\(?:\.?|--)(${opts.classNameRegExp})`, 'g');
```

If you are willing to introduce this, happy to create a PR with tests.

**Comment by sndyuk** [comment]
@tameraydin Thank you for the suggestion! It sounds good. But I'm not sure the proposed regex works fine since the original regex has `\\\.` and the new one doesn't.

**Comment by tameraydin** [comment]
@sndyuk you might be right, however while creating the PR I realized that none of your test cases will cover that branch (L22) but goes through L24 😬 (I guess this is because of the webpack test environment, due to style-loader perhaps; all of the files being processed are counting as JS) 

Created a new spec using the existing setup, so still no coverage for CSS files - I don't have time for it, but in my private repo (where I use exact same diff ^ that I suggested) things seem to be working fine 😅 
I also tested other existing cases having the new option enabled - they all pass.

**Comment by tameraydin** [comment]
Also FYI; npm warned that the lock-file needed to be updated, so I did that - and also fixed found vulnerabilities from npm audit, and deduped dependencies. (You can see each in separate commits, small diffs)

**Comment by sndyuk** [comment]
@tameraydin Thank you for the P/R! Checked the new regex deeply and there is something I don't understand. Please review the diff I made below and fix the code if your code works well with this.

1. 
> `\(?:\.?|--)`

Is this the first letter `\` required?

2. 
> `\(?:\.?|--)` 

The first dot(.) is necessary.  so ``\.?` should be `\.`(without `?`) ? Otherwise `.cl-test` and `url("cl-test.jpg")` are both matched.

3. 
> `(?:\\\.|--)`

Added one more `\` here. but I'm actually not sure what is the difference between `(?:\\\.|--)` and `(?:\.|--)`. There will be no difference, I guess. So let's make it be same as original one.

```diff
diff --git a/lib/optimizer.js b/lib/optimizer.js
index 9111ad1..a13579d 100644
--- a/lib/optimizer.js
+++ b/lib/optimizer.js
@@ -20,11 +20,11 @@ const optimize = (compiler, [file, originalSource], compilation, opts, classGene
   let classnameRegex;
   if (file.match(/.+\.css.*$/)) {
     classnameRegex = opts.mangleCssVariables
-      ? new RegExp(`\(?:\.?|--)(${opts.classNameRegExp})`, 'g')
+      ? new RegExp(`(?:\\\.|--)(${opts.classNameRegExp})`, 'g')
       : new RegExp(`\\\.(${opts.classNameRegExp})`, 'g');
   } else if (file.match(/.+\.js.*$/) || file.match(/.+\.html.*$/)) {
     classnameRegex = opts.mangleCssVariables
-      ? new RegExp(`\(?:["'\`.\\\s]|--)(${opts.classNameRegExp})`, 'g')
+      ? new RegExp(`(?:["'\`.\\\s]|--)(${opts.classNameRegExp})`, 'g')
       : new RegExp(`["'\`.\\\s](${opts.classNameRegExp})`, 'g');
   }
   if (!classnameRegex) {
```

**Comment by tameraydin** [comment]
Thanks for the review and pointing out that case - you are right! (Again, there needs to be some refactoring on the test setup in order to cover such CSS specific scenarios - currently they are all processed as JS content, and properties such as `url` or `content` are "expectedly" being mangled 🙃)

1 & 3 makes sense. Applied your suggestions (amended) - all tests are passing (with `mangleCssVariables` on and off).

**Comment by sndyuk** [comment]
@tameraydin I've just released the new version (5.1.0). 

> Again, there needs to be some refactoring on the test setup in order to cover such CSS specific scenarios - currently they are all processed as JS content, and properties such as url or content are "expectedly" being mangled 🙃

True... I'll create the issue!
